### PR TITLE
Support "unlink" functionality for iOS

### PR DESCRIPTION
**Pull request by grabbou**
Todo: 
- [X] remove project from Xcode
- [X] unit tests

### CODE REVIEW

File: src/ios/unlinkAssets.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      `Group 'Resources' does not exist in your XCode project. There is nothing to unlink.`
+    );
+  }
+
+  const plist = plistParser.parse(
+    fs.readFileSync(plistPath, 'utf-8')
+  );
+
+  const fonts = (assets.font || [])
+    .map(asset =>
```
Comment (Kureev) [inline src/ios/unlinkAssets.js]: mb `forEach`?

**Comment by Kureev** [comment]
Looks good so far! When you're planning to finish it?

**Comment by grabbou** [comment]
I hope today!

**Comment by grabbou** [comment]
Fixed eta: End of today

**Comment by Kureev** [comment]
new ETA? :laughing:

**Comment by grabbou** [comment]
I am having exams this week so going slower than I thought, but it's almost there :dancer:

**Comment by Kureev** [comment]
Can you rebase this branch?

**Comment by grabbou** [comment]
Will do right after I send over test suites :)

**Comment by grabbou** [comment]
Ready for review.

File: src/ios/addToHeaderSearchPaths.js [hunk 2]
```diff
@@ -1,29 +1,5 @@
...
-        .filter(path => path.indexOf('react-native/React/**'))
-        .length > 0;
+const mapHeaderSearchPaths = require('./mapHeaderSearchPaths');
 
-      if (shouldAddSearchPath) {
-        buildSettings.HEADER_SEARCH_PATHS.push(path);
-      }
-    });
+module.exports = function addToHeaderSearchPaths(project, path) {
+  mapHeaderSearchPaths(project, searchPaths => searchPaths.concat(path));
```
Comment (Kureev) [inline src/ios/addToHeaderSearchPaths.js]: Usually you return an array from the `map`-like functions

Comment (grabbou) [inline src/ios/addToHeaderSearchPaths.js]: Usually you pass an array to map, here we pass project and mapping happens inside it (check the `func` application. It's mutating because of underyling library design, so the only way to get over that was to keep that in a separate stuff.

File: src/link.js [hunk 3]
```diff
@@ -21,6 +21,50 @@ const getProjectDependencies = () => {
...
+    log.info(`Linking ${dependency.name} android dependency`);
+
+    const didLink = registerDependencyAndroid(
+      dependency.name,
+      dependency.config.android,
+      project.android
+    );
+
+    if (didLink) {
+      log.info(`Android module ${packageName} has been successfully linked`);
```
Comment (Kureev) [inline src/link.js]: if you notify a user in a happy scenario, you should do the same for the opposite one.

File: src/link.js [hunk 4]
```diff
@@ -21,6 +21,50 @@ const getProjectDependencies = () => {
...
+    }
+  }
+
+  if (project.ios && dependency.config.ios) {
+    log.info(`Linking ${dependency.name} ios dependency`);
+
+    if (registerDependencyIOS(dependency.config.ios, project.ios)) {
+      log.info(`iOS module ${packageName} has been successfully linked`);
```
Comment (Kureev) [inline src/link.js]: Same as above ^

File: src/link.js [hunk 5]
```diff
@@ -21,6 +21,50 @@ const getProjectDependencies = () => {
...
+  }
+
+  if (project.ios) {
+    log.info('Linking assets to ios project');
+    copyAssetsIOS(assets, project.ios);
+  }
+
+  if (project.android) {
+    log.info('Linking assets to android project');
+    copyAssetsAndroid(assets, project.android.assetsPath);
```
Comment (Kureev) [inline src/link.js]: If we introduce notifications about linking status, why don't we have them for the assets?

File: src/ios/addToHeaderSearchPaths.js (hunk 2 above)
Comment (Kureev) [inline src/ios/addToHeaderSearchPaths.js]: Mmmm, based on what you've said, I see a two options which can help us to move forward with this:
- Rename it to something like `patchHeaderSearchPaths`
- Add a comment that this function is impure and it mutates some state inside

Personally I prefer the first option, but I'll be fine with the second one also.

**Comment by Kureev** [comment]
Fix user notifications to make them consistent and we're ready to go :rocket:

File: src/link.js (hunk 3 above)
Comment (grabbou) [inline src/link.js]: I was just following `unlink` part (trying to match the style). Any changes e.g. `Android module already linked` should be subject to another PR.

**Comment by grabbou** [comment]
They are consistent :P But meh, I'll update both files anyway.

**Comment by Kureev** [comment]
I mean add them for failed linking and assets also :smiley:
